pkg/controller/pxc/replication.go

```diff
@@ -126,7 +126,9 @@ func (r *ReconcilePerconaXtraDBCluster) reconcileReplication(ctx context.Context
 	}
 
 	primary, err := r.getPrimaryPod(cr)
-	if err != nil {
+	if len(podList) == 1 && cr.Spec.PXC.Size == 1 && errors.Is(err, NoProxyDetectedError) {
+		primary = podList[0].Status.PodIP
+	} else if err != nil {
```

Comment:
Hello @gjreasoner  and thank you for your contribution.

Since `getPrimaryPod` is used in multiple places, I think it is better to introduce the logic of handling the case of not configured proxy inside that function so that we can have more consistency when it Is reused. WDYT?

Reply:
agree sounds like the right thing to do, ty

Reply:
@gjreasoner any updates for this one?

Reply:
@gjreasoner ping